### PR TITLE
Problem: recent dependencies features are not integrated

### PR DESCRIPTION
**Pull request by yihuang**
- integrate latest memiavl and local state-sync commands in cosmos-sdk

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 - [#984](https://github.com/crypto-org-chain/cronos/pull/984) experimental integration of memiavl.
 - [#985](https://github.com/crypto-org-chain/cronos/pull/985) Fix versiondb verify command on older versions
 - [#1038](https://github.com/crypto-org-chain/cronos/pull/1038) Update ibc-go to `v5.2.1`.
+- [#]() Integrate latest memiavl and cosmos-sdk
```
Comment (yihuang) [inline CHANGELOG.md:9]:
```suggestion
- [#1043](https://github.com/crypto-org-chain/cronos/pull/1043) Integrate latest memiavl and cosmos-sdk
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - [#984](https://github.com/crypto-org-chain/cronos/pull/984) experimental integration of memiavl.
 - [#985](https://github.com/crypto-org-chain/cronos/pull/985) Fix versiondb verify command on older versions
 - [#1038](https://github.com/crypto-org-chain/cronos/pull/1038) Update ibc-go to `v5.2.1`.
+- [#1043](https://github.com/crypto-org-chain/cronos/pull/1043) Integrate latest memiavl and cosmos-sdk
```
Comment (yihuang) [inline CHANGELOG.md:9]:
```suggestion
- [#1043](https://github.com/crypto-org-chain/cronos/pull/1043) Integrate latest memiavl and local state-sync commands in cosmos-sdk
```

File: app/config/config.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	// SnapshotInterval defines the block interval the memiavl snapshot is taken, default to 1000.
+	SnapshotInterval uint32 `mapstructure:"snapshot-interval"`
+	// CacheSize defines the size of the cache for each memiavl store.
+	CacheSize int `mapstructure:"cache-size"`
+}
+
+func DefaultMemIAVLConfig() MemIAVLConfig {
+	return MemIAVLConfig{
+		CacheSize:        DefaultCacheSize,
+		SnapshotInterval: memiavl.DefaultSnapshotInterval,
```
Comment (mmsqe) [inline app/config/config.go:36]: seems miss some change like [ZeroCopy](https://github.com/crypto-org-chain/cronos/pull/1035/files#diff-2a2301646ea40f7638775b9a0cfbd0cb8c9f081b5f8732a73f44767aeae95e9dR37)

Comment (yihuang) [reply app/config/config.go:36]: good cache, fixed.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**